File: playground/steps/basic-tutorial/services/pet-store.ts
import { Order, Pet } from './types'

export const petStoreService = {
  createPet: async (pet: Omit<Pet, 'id'>): Promise<Pet> => {
    const response = await fetch('https://petstore.swagger.io/v2/pet', {
      method: 'POST',
      body: JSON.stringify({
        name: pet.name,
        photoUrls: [pet.photoUrl],
        status: 'available',
      }),
      headers: { 'Content-Type': 'application/json' },
    })
    return response.json()
  },
  createOrder: async (order: Omit<Order, 'id'>): Promise<Order> => {
    const response = await fetch('https://petstore.swagger.io/v2/store/order', {
      method: 'POST',
      body: JSON.stringify(order),
      headers: { 'Content-Type': 'application/json' },
    })
    return response.json()
  },
}

File: playground/steps/default_python/services/pet_store.py
import httpx
from typing import Dict, Any
from .types import Order, Pet

class PetStoreService:
    async def create_pet(self, pet: Dict[str, Any]) -> Pet:
        pet_data = {
            "name": pet["name"],
            "photoUrls": [pet["photo_url"]],
            "status": "available"
        }
        
        async with httpx.AsyncClient() as client:
            response = await client.post(
                'https://petstore.swagger.io/v2/pet',
                json=pet_data,
                headers={'Content-Type': 'application/json'}
            )
            return response.json()
    
    async def create_order(self, order: Dict[str, Any]) -> Order:
        async with httpx.AsyncClient() as client:
            response = await client.post(
                'https://petstore.swagger.io/v2/store/order',
                json=order,
                headers={'Content-Type': 'application/json'}
            )
            return response.json()

pet_store_service = PetStoreService()